telemetry/specs/INT.mdk

```diff
@@ -724,22 +757,31 @@ INT shim header for TCP/UDP:
     0                   1                   2                   3
     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-   |      Type     |    Reserved   |     Length    |   DSCP    |R R|
+   | Type  |NPT|R|R|     Length    |  UDP port, IP Proto, or DSCP  |
    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 `
 
-* Type: This field indicates the type of INT Header following the shim header.
+* Type (4b): This field indicates the type of INT Header following the shim header.
 Two Type values are used: one for the hop-by-hop header type and the other for
 the destination header type (See Section [#sec-int-header-types]).
-* Length: This is the total length of INT metadata header, INT stack and the
+* NPT (Next Protocol Type, 2b): This field is meaningful only when the UDP destination
+port number (INT_TBD) is used to indicate the existence of INT. In the other cases,
+this field must be zero. When UDP destination port is INT_TBD, this field may have
+one of the two values:
+** one (1) indicates that the original UDP payload is followed by the INT stack,
+and the last two bytes of the shim header carries the original UDP destination port.
+** two (2) indicates that another (the original) L4 header is followed by
+the INT stack, and the last byte of the shim header carries the IP protocol
+value for the L4 layer.
+* Length (8b): This is the total length of INT metadata header, INT stack and the
 shim header in 4-byte words. A non-INT device may read this field
 and skip over INT headers.
-
-* DSCP: If IP DSCP is used to indicate INT, this field optionally
-stores the original DSCP value. Otherwise, this field
+* UDP port, IP proto, or DSCP: If IP DSCP is used to indicate INT, this field optionally
+stores the original DSCP value. For non-zero values of NPT, either UDP port number
+or IP protocol number is carried accordingly. Otherwise, this field
 is reserved.
```

Comment:
This text is not so clear. I prefer to spell it out:

UDP port, IP proto, or DSCP: The contents of this field differ depending on the value of NTP:
NTP 0: The first byte and the last two bits are reserved, set to zero upon transmission and ignored upon reception. The second byte may optionally carry the original DSCP value.
NTP 1: The original UDP destination port value.
NTP 2: The first byte is reserved, set to zero upon transmission and ignored upon reception. The second byte carries the original IP protocol value.

Reply:
Yes